pubpub/pubpub
# Submission page state updates

This PR moves state management for the Spub page out of the `SpubHeader` and up several levels, so `{ submissionState, updateSubmissionState }` can be accessed from the `usePubContext()` hook across the entire Pub page. It gets us closer to desired behavior in several ways:

- When the Instructions tab is shown, the Pub body is hidden
- When the Preview tab is shown, the current abstract appears within the Pub
- The abstract is properly appended to the Pub draft on the server side. My initial idea for how to approach this on the client had major problems — the `Editor` instance that's visible when the Preview is shown should be read-only and thus not a good vector for manipulating the draft via the collab plugin.

I have a follow-up coming along that fixes some other issues with Pub page state management that have cropped up here.

_Test plan:_
- Create a new Spub
- Verify that when you load the Spub, the instructions tab is visible and the rest of the Pub is hidden
- Verify that when you change the abstract, it's reflected in the Preview
- Verify that the abstract is attached when you submit the Spub
